### PR TITLE
fix(animations): getAnimationStyle causes exceptions in older browsers

### PR DESCRIPTION
**Pull request by Serginho**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #24094

## What is the new behavior?
Animation doesn't break in old browsers now 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by matsko** [comment]
@Serginho sorry this took us forever to get back to you on.

Could you add a test to this? You would need to export the function you modified and import it into https://github.com/angular/angular/blob/master/packages/animations/browser/test/render/css_keyframes/element_animation_style_handler_spec.ts

Then test by mocking out the a fake element a style property.

**Comment by Serginho** [comment]
@matsko I didn't do any additional test because `getAnimationStyle` is already tested, for example here:

https://github.com/angular/angular/blob/5bc39f8c8d5238a9be9bd968cf18ea4b738bd6be/packages/animations/browser/test/render/css_keyframes/element_animation_style_handler_spec.ts#L66-L68

So I assume you want a testcase for this bug, and this is what I did in the updated commit. 
The fix is squashed, formatted, and rebased to master.

Let me know if you need something else.

**Review by petebacondarwin — APPROVED** [review]
LGTM - please can you add some more information to the commit message body?

**Comment by petebacondarwin** [comment]
@Serginho - would you mind rebasing this PR on top of master. Then I will try to chase reviewers to get it merged.

**Comment by Serginho** [comment]
@petebacondarwin Sure. 

Rebased and ready to merge.

File: packages/animations/browser/test/render/css_keyframes/element_animation_style_handler_spec.ts [hunk 1]
```diff
@@ -227,5 +226,23 @@ const EMPTY_FN = () => {};
...
+        style: {
+          'animationstyle1': 'value',
+          'animationstyle2': null,
+          'animationstyle3': '',
+          'animation': null
+        }
+      };
+      expect(getAnimationStyle(fakeElement, 'style1')).toBe('value');
+      expect(getAnimationStyle(fakeElement, 'style2')).toBe('');
+      expect(getAnimationStyle(fakeElement, 'style2')).toBe('');
```
Comment (petebacondarwin) [inline packages/animations/browser/test/render/css_keyframes/element_animation_style_handler_spec.ts:244]: should this be `style3`?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Thanks for rebasing @Serginho. Just one minor correction in the test.

File: packages/animations/browser/test/render/css_keyframes/element_animation_style_handler_spec.ts (hunk 1 above)
Comment (Serginho) [reply packages/animations/browser/test/render/css_keyframes/element_animation_style_handler_spec.ts:244]: oops

**Review by petebacondarwin — APPROVED** [review]
Great! Thanks.

**Review by crisbeto — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:343086889:BASE:343086898:1605718003172:1d70e0f7).

File: packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts [hunk 2]
```diff
@@ -140,8 +140,8 @@ function setAnimationStyle(element: any, name: string, value: string, index?: nu
   element.style[prop] = value;
 }
 
-function getAnimationStyle(element: any, name: string) {
-  return element.style[ANIMATION_PROP + name];
+export function getAnimationStyle(element: any, name: string) {
+  return element.style[ANIMATION_PROP + name] || '';
```
Comment (AndrewKushnir) [inline packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts:144]: Quick question: is it possible to have `0` (as integer) or `false` as a value here? If yes, we should probably use nullish coalescing operator instead:
```suggestion
  return element.style[ANIMATION_PROP + name] ?? '';
```

Comment (AndrewKushnir) [reply packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts:144]: Just checked the code that invokes this function and it looks like the string value is expected, so the current change (`|| ''`) is correct.

Comment (Serginho) [reply packages/animations/browser/src/render/css_keyframes/element_animation_style_handler.ts:144]: One year ago when I developed this, nullish coalescing didn't exists in the current angular version 😄. I remember that I checked CSSStyleDeclaration animation property was declared as string.
See: https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L2763

**Comment by AndrewKushnir** [comment]
@Serginho this PR is now merged, thanks for contributing to Angular!

**Pull request closed without merging**